tests/models/test_hparams.py

```diff
@@ -82,14 +85,16 @@ def test_omega_conf_hparams(tmpdir, cls):
     # init model
     conf = OmegaConf.create(dict(test_arg=14, mylist=[15.4, dict(a=1, b=2)]))
     model = cls(hparams=conf)
+    assert isinstance(model.hparams, Container)
 
     # run standard test suite
     raw_checkpoint_path = _run_standard_hparams_test(tmpdir, model, cls)
-    model = cls.load_from_checkpoint(raw_checkpoint_path)
+    model2 = cls.load_from_checkpoint(raw_checkpoint_path)
+    assert isinstance(model2.hparams, Container)
 
     # config specific tests
-    assert model.hparams.test_arg == 14
-    assert model.hparams.mylist[0] == 15.4
+    assert model2.hparams.test_arg == 14
+    assert model2.hparams.mylist[0] == 15.4
```

Comment:
maybe
```python
assert model2.hparams == model2
```

Reply:
1. Please confirm model2.hparams type is identical to the input type (DictConfig in -> DictConfig out).
2. Since this is used to restore from checkpoint, I recommend using cloudpickle to serialize and deserialize it.
If you want a user readable version near it you can also dump the yaml file.
The reason is that OmegaConf 2.0 (used by Hydra 1.0 which is a release candidate now) adds runtime type safety: going through a yaml file will lose this functionality.

Reply:
what is the advantage of `cloudpickle` to building `pickle`?

Reply:
I remember some issues serializing Python types with pickle which seems okay in a simple test I just ran across both cloudpickle and pickle ion 3.6, 3.7 and 3.8.
So never mind. Both pickle and cloudpickle seems okay.

```python
from dataclasses import dataclass
import cloudpickle
import pickle
from omegaconf import OmegaConf, MISSING, ValidationError
import pytest


@dataclass
class User:
    name: str = MISSING
    age: int = MISSING


def _test_cfg(c):
    assert OmegaConf.get_type(c) is User
    with pytest.raises(ValidationError):
        c.age = "oops"  # runtime validation error


@pytest.mark.parametrize("serializer", [pickle, cloudpickle])
def test_serialize(serializer):
    cfg1 = OmegaConf.structured(User(name="James Bond", age=7))
    cfg2 = serializer.loads(serializer.dumps(cfg1))
    assert cfg1 == cfg2
    _test_cfg(cfg1)
    _test_cfg(cfg2)
```

Reply:
if there is a reason we can change it in another PR or make it as an optional default
```python
try:
    import cloudpickle as pickle
except ImportError:
    import pickle
```

Reply:
As of right now, I am not aware of any issues with pickle serializing OmegaConf objects.

I am not testing cloudpickle in OmegaConf right now, but I will add that as a test-only dependency.
It might want to do there same, or not test it at all for now.